### PR TITLE
Add description of protobuf and clarify versioning policy

### PR DESCRIPTION
**Pull request by smarterclayton**
Fixes kubernetes/kubernetes#8132

### CODE REVIEW

**Comment by wojtek-t** [comment]
Not sure what's the procedure in this repo. Can I merge it?

**Comment by erictune** [comment]
@wojtek-t is this documentation relevant to 1.2?   
If not, please close this PR, and rebase your work on top of release-1.3, and open a new PR.

To rebase your work on 1.3 do something like this starting in the branch that you used for this PR:
```
COMMITS=$(git log master.. | grep commit | cut -f 2 -d " " | tac)
BR=$(git rev-parse --abbrev-ref HEAD)
git fetch upstream
git checkout upstream/release-1.3
git checkout -b $BR-1.3
for c in $COMMITS
do
  git cherry-pick $c; done
done
git push
```

**Comment by wojtek-t** [comment]
@erictune - this is documentation for 1.3 release.
Do we really consider master as "1.2-related"? This is super strange, given that we are releasing 1.3 in few days.

**Comment by johndmulhausen** [comment]
@wojtek-t yes, the master branch is for 1.2 docs only until we promote release-1.3 to be the default site contents. Also please don't add Docs LGTM labels without a more thorough understanding of where certain files live; the link to the proposals document will not resolve.

**Comment by johndmulhausen** [comment]
@smarterclayton please re-open this PR against the release-1.3 branch as @erictune has instructed.

File: docs/api.md [hunk 1]
```diff
@@ -25,13 +25,15 @@ Complete API details are documented using [Swagger v1.2](http://swagger.io/). Th
 
 We also host a version of the [latest API documentation UI](http://kubernetes.io/kubernetes/third_party/swagger-ui/). This is updated with the latest release, so if you are using a different version of Kubernetes you will want to use the spec from your apiserver.
 
+Kubernetes implements an alternative Protobuf based serialization format for the API that is primarily intended for intra-cluster communication, documented in the [design proposal](proposals/protobuf.md) and the IDL files for each schema are located in the Go packages that define the API objects.
```
Comment (johndmulhausen) [inline docs/api.md]: URL should be: https://github.com/kubernetes/kubernetes/blob/{{ page.githubbranch }}/docs/proposals/protobuf.md

**Comment by smarterclayton** [comment]
I already did in https://github.com/kubernetes/kubernetes.github.io/pull/669, someone closed it.  Which one is it?

**Comment by smarterclayton** [comment]
It sounds like we really want #669 - reopening and closing this one (moving the comments).

**Comment by smarterclayton** [comment]
Sigh, wrong PR.

**Pull request closed without merging**